### PR TITLE
[PageActions]Remove border top

### PR DESCRIPTION
**Pull request by heyjoethomas**
Remove top border from `PageActions` as it is almost invisible

### WHY are these changes introduced?

Fixes [#958](https://github.com/orgs/Shopify/projects/7136/views/14?pane=issue&itemId=33496490)

### WHAT is this pull request doing?

Removes top border and condenses spacing of `PageActions`

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: .changeset/sixty-bags-wave.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': minor
+---
+
+Remove top border from PageActions as it is almost invisible with uplift styles
```
Comment (sophschneider) [inline .changeset/sixty-bags-wave.md]: We can remove this! ware are only adding changesets for non-experimental styles.

File: polaris-react/src/components/PageActions/PageActions.scss [hunk 2]
```diff
@@ -2,10 +2,9 @@
 
 .PageActions {
   margin: 0 auto;
-  padding: var(--p-space-5);
-  border-top: var(--p-border-width-1) solid var(--p-color-border-subdued);
+  padding: var(--p-space-4);
```
Comment (sophschneider) [inline polaris-react/src/components/PageActions/PageActions.scss]: We should put this behind the beta 

```suggestion
  padding: var(--p-space-5);
  border-top: var(--p-border-width-1) solid var(--p-color-border-subdued);
  
  #{$se23} & {
    padding: var(--p-space-4);
  }
```

File: polaris-react/src/components/PageActions/PageActions.scss [hunk 3]
```diff
@@ -2,10 +2,9 @@
...
 
   @media #{$p-breakpoints-sm-up} {
-    padding: var(--p-space-5) 0;
+    padding: var(--p-space-4) 0;
```
Comment (sophschneider) [inline polaris-react/src/components/PageActions/PageActions.scss]:
```suggestion
    padding: var(--p-space-5) 0;
  
    #{$se23} & {
       padding: var(--p-space-4) 0;
     }
```

**Review by sophschneider — APPROVED** [review]
🎩'd on storybook, LGTM!

**Review by lgriffee — APPROVED** [review]
✨

**Pull request merged**